documentation/1.2/tour/interop.md

```diff
@@ -85,7 +85,107 @@ Aether. You can find more information [here](../../reference/repository/maven).
 
 ## Deploying Ceylon on OSGi
 
-_TODO: document OSGi interop._
+Ceylon is fully interoperable with OSGI, so that Ceylon modules:
+- can be deployed as pure OSGI bundles in an OSGI container out-of-the-box without any modification of the module archive file,
+- can embed additional OSGI metadata, to declare services for example,
+- can easily use OSGI standard services
+
+This provides a great and straightforward opportunity to run Ceylon code inside a growing number of JEE application servers or enterprise containers that are base upon (or integrated with) OSGI.
+
+#### Installing the Ceylon distribution / Ceylon SDK in an OSGI container
+
+In order to be able to resolve and start Ceylon module archives (`.car` files) inside an OSGI container, you will first need to install, in the OSGI container, all the bundles of the Ceylon distribution and SDK.
+
+These bundles are available in a dedicated location on the Ceylon language web site under various delivery forms:
+- OSGI bundle repositories (OBR and R5 XML), for Felix-based containers for example,
+- P2 repositories for Eclipse development or deployment to an Equinox container,
+- Zip archives for direct deployment inside containers,
+- Apache Karaf (aka JBoss Fuse) [features](http://karaf.apache.org/manual/latest/users-guide/provisioning.html)
+
+The OSGI interoperability reference gives more details about the [URLs providing these packages](../../reference/interoperability/osgi.md#retrieving-the-ceylon-distribution-and-sdk-for-osgi), as well as [how to install](../../reference/interoperability/osgi.md#installing-the-ceylon-distribution-and-sdk-in-an-osgi-container) these bundles, for various containers
+
+#### OSGI Metadata management
+
+[Module archives](./modules#module_archives_and_module_repositories) generated
+for execution on the Java virtual machine already contain the OSGI metadata that
+can be deduced from the Ceylon module descriptor.
+
+More precisely, for the following module descriptor:
+```Ceylon
```

Comment:
This code block doesn’t work, probably also needs blank lines before and after the ```` ``` ````.